README.md

```diff
@@ -1,85 +1,61 @@
 [![License](https://img.shields.io/hexpm/l/hammer.svg)](https://github.com/ExHammer/hammer/blob/master/LICENSE.md)
 
-A rate-limiter for Elixir, with pluggable storage backends.
-
-## Hammer-Plug
-
-We have a helper-library to make adding rate-limiting to your Phoenix
-(or other plug-based) application even easier: [Hammer.Plug](https://github.com/ExHammer/hammer-plug).
+**Hammer** is a rate-limiter for Elixir with pluggable storage backends. Hammer enables users to set limits on actions performed within specified time intervals, applying per-user or global limits on API requests, file uploads, and more.
 
 ## Installation
 
-Hammer is [available in Hex](https://hex.pm/packages/hammer), the package can be installed
-by adding `:hammer` to your list of dependencies in `mix.exs`:
+Hammer is [available in Hex](https://hex.pm/packages/hammer). Install by adding `:hammer` to your list of dependencies in `mix.exs`:
 
 ```elixir
 def deps do
   [
-    {:hammer, "~> 6.1"}
+    {:hammer, "~> 7.0"}
   ]
 end
 ```
 
-## Documentation
+## Default Algorithm
 
-On HexDocs: [https://hexdocs.pm/hammer/frontpage.html](https://hexdocs.pm/hammer/frontpage.html)
+By default, Hammer uses a **fixed window counter** to track actions within set time windows, resetting the count at the start of each new window. For example, with a limit of 10 uploads per minute, a user could upload up to 10 files between 12:00:00 and 12:00:59, and up to 10 more between 12:01:00 and 12:01:59.
 
-The [Tutorial](https://hexdocs.pm/hammer/tutorial.html) is an especially good place to start.
+## Core Concepts
 
-## Usage
+- **Limit:** Maximum number of actions allowed in a window.
+- **Scale:** Duration of the time window (in milliseconds).
+- **Key:** Unique identifier (e.g., user ID) to scope the rate limiting.
 
-Example:
+## Example Usage
 
 ```elixir
-defmodule MyApp.VideoUpload do
-
-  def upload(video_data, user_id) do
-    case Hammer.check_rate("upload_video:#{user_id}", 60_000, 5) do
-      {:allow, _count} ->
-        # upload the video, somehow
-      {:deny, _limit} ->
-        # deny the request
-    end
-  end
-
+defmodule MyApp.RateLimit do
+  use Hammer, backend: :ets
 end
-```
 
-The `Hammer` module provides the following functions:
+MyApp.RateLimit.start_link()
 
-- `check_rate(id, scale_ms, limit)`
-- `check_rate_inc(id, scale_ms, limit, increment)`
-- `inspect_bucket(id, scale_ms, limit)`
-- `delete_buckets(id)`
+user_id = 42
+key = "upload_video:#{user_id}"
+scale = :timer.minutes(1)
+limit = 3
 
-Backends are configured via `Mix.Config`:
-
-```elixir
-config :hammer,
-  backend: {Hammer.Backend.ETS, [expiry_ms: 60_000 * 60 * 4,
-                                 cleanup_interval_ms: 60_000 * 10]}
+case MyApp.RateLimit.check_rate(key, scale, limit) do
+  {:allow, _count} -> :upload_the_video
+  {:deny, _limit} -> :deny_the_request
+end
 ```
 
-See the [Tutorial](https://hexdocs.pm/hammer/tutorial.html) for more.
```

Comment:
what s wrong with the tutorial link? we should keep it?

Reply:
I think tutorial can be part of the README, no need to make users look for it, click the links.

Reply:
sure.